pandas/core/frame.py

```diff
@@ -732,7 +732,7 @@ def __init__(
                 data, axes={"index": index, "columns": columns}, dtype=dtype, copy=copy
             )
 
-        elif isinstance(data, dict):
+        elif isinstance(data, abc.MutableMapping):
```

Comment:
I believe this can be 
```suggestion
        elif isinstance(data, abc.Mapping):
```
instead, as it just need to access the dict, so, anything that behaves like the dict access can be passed.